networktable.c

```diff
@@ -41,17 +41,101 @@ bool create_network_table(void)
 bool create_network_addresses_table(void)
 {
 	bool ret;
+
+	dbquery("BEGIN TRANSACTION");
+
 	// Create network_addresses table in the database
 	ret = dbquery("CREATE TABLE network_addresses ( network_id INTEGER NOT NULL, "\
 	                                               "ip TEXT NOT NULL, "\
-	                                               "lastSeen INTEGER NOT NULL, "\
+	                                               "lastSeen INTEGER NOT NULL DEFAULT (cast(strftime('%%s', 'now') as int)), "\
 	                                               "UNIQUE(network_id,ip), "\
 	                                               "FOREIGN KEY(network_id) REFERENCES network(id));");
-	if(!ret){ dbclose(); return false; }
+	if(!ret)
+	{
+		logg("create_network_addresses_table(): CREATE TABLE network_addresses failed!");
+		dbclose();
```

Comment:
Should we also run a `ROLLBACK` here?

Reply:
If `CREATE TABLE` failed, it shouldn't have done anything.

Reply:
Maybe for the later statements then.

Reply:
It looks like SQLite will close the transaction for you in this case.

Reply:
Quoting the SQLite documentation:

> Transactions can be started manually using the BEGIN command. Such transactions usually persist until the next COMMIT or ROLLBACK command. But a transaction will also ROLLBACK if the database is closed or if an error occurs and the ROLLBACK conflict resolution algorithm is specified. See the documentation on the ON CONFLICT clause for additional information about the ROLLBACK conflict resolution algorithm. 

We use the standard which is `ON CONFLICT ABORT`:

> When an applicable constraint violation occurs, the ABORT resolution algorithm aborts the current SQL statement with an SQLITE_CONSTRAINT error and backs out any changes made by the current SQL statement